### ISSUE TITLE
One-shot use: output nix-store options

### ISSUE THREAD

**Issue by 7c6f434c** [cachix/cachix#126]
Sometimes I want just a single path from a Cachix cache without a plan to use Cachix by default.

In such a case it would be nice to say something like `cachix use cache-name --one-shot` and get the output like:
```
  --option binary-caches \
      'https://cache.nixos.org/ https://cache-name.cachix.org/' \
  --option trusted-public-keys  '
     cache-name.cachix.org-1:00000000000000000000000000000000=
     cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```

Also, both `--nixos` and `--one-shot` could be mentioned by `--help`…

My personal interest is related to having `nix.conf` as a symlink to store on a not-literally-NixOS system.

**Comment by domenkozar** [comment]
@7c6f434c would this be similar to https://github.com/cachix/cachix/issues/14 - where you'd use cachix <name> only for specific project?

**Comment by 7c6f434c** [comment]
Probably yes, but hopefully cheaper to implement and maybe a bit more straightforward for use in wrapper scripts.

**Comment by domenkozar** [comment]
Sounds reasonable. About naming, I was thinking: `--print-nixos` and `--print-nix`.

**Comment by 7c6f434c** [comment]
I have no objections about naming if `--help` lists these flags.

**Comment by domenkozar** [comment]
It does list existing `--nixos`:

```
 cachix use --help
Usage: cachix use NAME [-n|--nixos]
  Configure nix.conf to enable binary cache during builds

Available options:
  -h,--help                Show this help text
  -n,--nixos               Output NixOS configuration lines

ExitSuccess
```

**Comment by 7c6f434c** [comment]
Oh right, subcommands have their own help entries.

Interesting that `--host` is listed for `cachix --help` but not `cachix use --help`. And in `cachix --help` there are two conflicting interpretations of `-h`.

**Comment by domenkozar** [comment]
Yes it's confusing, `--help` on top-level command lists global options and those are not repeated for each subcommand, which lists options only for itself.

**Comment by roberth** [comment]
That's [an issue in optparse-applicative](https://github.com/pcapriotti/optparse-applicative/issues/294)

**Comment by domenkozar** [comment]
And the other issue is https://github.com/pcapriotti/optparse-applicative/issues/175

**Comment by domenkozar** [comment]
I gave this a go in a704c7a4cece9ad94e41b1e8d5d2cd700ece9a78, but it's not really feasible to implement until https://github.com/NixOS/nix/issues/2149 is implemented in Nix.
